src/Operations.jl

```diff
@@ -1298,4 +1299,16 @@ function test(ctx::Context, pkgs::Vector{PackageSpec}; coverage=false, test_fn=n
     end
 end
 
+function package_info(ctx::Context, pkg::PackageSpec;
+                      name=false, version=false, pinned=false, developed=false, source=false
+                      )::PackageInfo
+    info = PackageInfo()
+    name      && (info.name = pkg.name)
+    version   && (info.version = (pkg.version == VersionSpec() ? nothing : pkg.version))
+    pinned    && (info.pinned = pkg.pinned)
+    source    && (info.source = project_rel_path(ctx, source_path(pkg)))
+    developed && (info.developed = pkg.path !== nothing)
+    return info
+end
```

Comment:
Actually Package info can be made immutable fairly easily.
I think it would be good to do so, since it would be very confusing if someone mutated it in their code.

```
info = PackageInfo(
    name      = name ? pkg.name : nothing,
    version   = version && pkg.version == VersionSpec() ? pkg.version : nothing,
    pinned    = pinned ? pkg.pinned : nothing,
    source    = source ? project_rel_path(ctx, source_path(pkg)) : nothing,
    developed = developed ? pkg.path !== nothing : nothing,
)
return info
```